### ISSUE TITLE
Using Semantic-UI sidebar with Angular 2

### ISSUE THREAD

**Issue by peb7268** [Semantic-Org/Semantic-UI#4877]
So I know there are some dupe issues floating around about the following error messages: 
*Sidebar: Had to add pusher element. For optimal performance make sure body content is inside a pusher element*  
*Sidebar: Had to move sidebar. For optimal performance make sure sidebar and pusher are direct children of your body tag *

I think my use case is a bit different than those though. 

I can't make my sidebars direct children of the body tag since they need to exist in an angular root element and making 3 separate ngModules ( 1 for the app, and one for each of my two sidebars ) would be wildly inefficient and problematic in this case ). 

That leaves me with solution of trying to use a **custom context** to initialize my sidebar.  I tried that solution and it doesn't work still. It tells me it keeps having to move the sidebar to be more efficient. 

Attached is a screenshot of my DOM structure.

<img width="745" alt="screen shot 2017-01-02 at 4 24 55 pm" src="https://cloud.githubusercontent.com/assets/1199718/21596358/39d338b2-d108-11e6-880a-0124baaa41b7.png">

**Comment by IonutBajescu** [comment]
Can you also show me the javascript bit that initializes the sidebar?

**Comment by peb7268** [comment]
Sigur ca pot! 

I currently have this: 

```
toggleProjectSideBar(){
    $('.left.sidebar')
    .sidebar('setting', 'transition', 'overlay')
    .sidebar('toggle');

    window.setTimeout(function(){
      let input:any = $('.left.sidebar input');
      input.focus();
    }, 500);
  }
  
  toggleHomeSidebar(){
    $('.home.sidebar')
    .sidebar('setting', 'transition', 'overlay')
    .sidebar('toggle');
  }
```

but I also tried: 

```
$('.left.sidebar')
    .sidebar({
        "context": "app-root app-home .admin-wrapper projects-sidebar"
    })
    .sidebar('toggle');
```

**Comment by Banandrew** [comment]
Hi @peb7268,

Thank you for posting, but although it’s a valid usage question, we’ve limited GitHub Issues to bug reports and feature requests, keeping the board more manageable for maintainers; see the [contributing guidelines](https://github.com/Semantic-Org/Semantic-UI/blob/master/CONTRIBUTING.md) for more information on what kind of posts should find themselves into the GitHub Issues board.

To get answers or feedback that might allow you to repost this issue, please use one of our other support resources:
* [Gitter chatroom](https://gitter.im/Semantic-Org/Semantic-UI)
* [Official forums](http://forums.semantic-ui.com)
* [StackOverflow](http://stackoverflow.com/questions/tagged/semantic-ui)

**Comment by peb7268** [comment]
Haha @banandrew what a BS answer. Aka we don't want to out in thought or development to support valid use cases of our product. #lame. It's cool. I already solved it wo the "official" way. I'll use bootstrap next time.

**Comment by Banandrew** [comment]
@peb7268 I’m sorry to hear that. I actually encountered a very similar problem with Vue 2 a couple of months ago that I also had to solve as an engineer by manipulating the sidebar’s context—in the end, that’s what we’re paid to do. I labeled the issue as a usage question, because you didn’t request any specific features, and only outlined a problem that had more to do with Angular than SUI—I didn’t know if you solved it, and I couldn’t see a way of helping you without thinking that the sidebar module will have to be somehow rewritten; the changes of that scale might not be feasible right now, as Jack’s currently has his hands full.

What I meant, is that you’d probably have more luck asking about it in an Angular community, and that it’s much easier to deal with the problem on the level of a JavaScript framework. If you tell me how you solved it, we might be able to figure out a way to file it as a proper feature request. Did you have any luck with the context property?
